core/src/layers/prometheus.rs

```diff
@@ -399,80 +243,187 @@ impl PrometheusLayerBuilder {
     pub fn register(self, registry: &Registry) -> Result<PrometheusLayer> {
-        let labels = OperationLabels::names(false, self.path_label_level > 0);
-        let operation_duration_seconds = HistogramVec::new(
-            histogram_opts!(
-                observe::METRIC_OPERATION_DURATION_SECONDS.name(),
-                observe::METRIC_OPERATION_DURATION_SECONDS.help(),
-                self.operation_duration_seconds_buckets
-            ),
-            &labels,
-        )
-        .map_err(parse_prometheus_error)?;
-        let operation_bytes = HistogramVec::new(
-            histogram_opts!(
-                observe::METRIC_OPERATION_BYTES.name(),
-                observe::METRIC_OPERATION_BYTES.help(),
-                self.operation_bytes_buckets
-            ),
-            &labels,
-        )
-        .map_err(parse_prometheus_error)?;
-
-        let labels = OperationLabels::names(true, self.path_label_level > 0);
-        let operation_errors_total = GenericCounterVec::new(
-            Opts::new(
-                observe::METRIC_OPERATION_ERRORS_TOTAL.name(),
-                observe::METRIC_OPERATION_ERRORS_TOTAL.help(),
-            ),
-            &labels,
-        )
-        .map_err(parse_prometheus_error)?;
-
-        let labels = OperationLabels::names(false, false);
-        let http_request_duration_seconds = HistogramVec::new(
-            histogram_opts!(
-                observe::METRIC_HTTP_REQUEST_DURATION_SECONDS.name(),
-                observe::METRIC_HTTP_REQUEST_DURATION_SECONDS.help(),
-                self.http_request_duration_seconds_buckets
-            ),
-            &labels,
-        )
-        .map_err(parse_prometheus_error)?;
-        let http_request_bytes = HistogramVec::new(
-            histogram_opts!(
-                observe::METRIC_HTTP_REQUEST_BYTES.name(),
-                observe::METRIC_HTTP_REQUEST_BYTES.help(),
-                self.http_request_bytes_buckets
-            ),
-            &labels,
-        )
-        .map_err(parse_prometheus_error)?;
-
-        registry
-            .register(Box::new(operation_duration_seconds.clone()))
-            .map_err(parse_prometheus_error)?;
-        registry
-            .register(Box::new(operation_bytes.clone()))
-            .map_err(parse_prometheus_error)?;
-        registry
-            .register(Box::new(operation_errors_total.clone()))
-            .map_err(parse_prometheus_error)?;
-        registry
-            .register(Box::new(http_request_duration_seconds.clone()))
-            .map_err(parse_prometheus_error)?;
-        registry
-            .register(Box::new(http_request_bytes.clone()))
-            .map_err(parse_prometheus_error)?;
+        let labels = OperationLabels::names();
+
+        let operation_bytes = {
+            let metric = observe::MetricValue::OperationBytes(0);
+            register_histogram_vec_with_registry!(
+                metric.name(),
+                metric.help(),
+                &labels,
```

Comment:
`operation_bytes` doesn't have label `error` and `status_code`, do we need to handle this?

Reply:
Yes, we need to do it, I forgot it